compiler/rustc_mir_transform/src/known_panics_lint.rs

```diff
@@ -897,8 +885,8 @@ impl CanConstProp {
         for (local, val) in cpv.can_const_prop.iter_enumerated_mut() {
             let ty = body.local_decls[local].ty;
             if ty.is_union() {
-                // Do not const prop unions as they can
-                // ICE during layout calc
+                // Unions are incompatible with const prop because
+                // Rust has no conpcept of the active variant of a union
```

Comment:
Also improved this comment while I was at it. The new comment provides a more fundamental reason why we can't const prop unions

Reply:
Typo: conpcept

Reply:
Whoops

Reply:
Fixed